packages/forms-web-app/src/utils/timetables/get-timetables-state.js

```diff
@@ -13,19 +17,30 @@ const getOpenEventDeadlineTimetables = (timetables) =>
 		isTimetableTypeOfEventDeadlineOpen(typeOfEvent, dateOfEvent, dateTimeDeadlineStart)
 	);
 
-const getHasOpenTimetables = async (case_ref) => {
-	const { data } = await getTimetables(case_ref);
-	let response = false;
-	if (data && data.timetables) {
-		const openEventDeadlineTimetables = getOpenEventDeadlineTimetables(data.timetables);
-		response = openEventDeadlineTimetables.length > 0;
+const getAllTimetables = async (session, case_ref) => {
+	let allTimetables = getTimetablesSession(session, case_ref);
+
+	if (!allTimetables) {
+		const { data } = await getTimetables(case_ref);
+
+		allTimetables = data?.timetables || [];
+
+		setTimetablesSession(session, case_ref, allTimetables);
```

Comment:
I think this may lead to an unwanted side effect.  
If I'm understanding this we will be storing the timetable for every user for every case they visit.  We may run the risk of running out of storage in redis if there is a spike.

I may be wrong, but I think this introduces the storage of the timetable data in session much earlier.

Reply:
I see your point. We could have an issue when users visit multiple projects which would cause all the timetables to be added to a project case reference key. We could end up with a large amount of data.

The reasoning for saving the data to the caseRef was to avoid having to create more complexity in the logic for detecting whether or not we have already checked the project. This way, if the check is undefined we know we need to do another api call.

As an alternative, we could only save a boolean value against each project, which would mean a much smaller object. Let me know you can foresee any potential issues with that method.

Reply:
I really like the simple boolean method if possible.  
In most cases current implementation would be fine but we do introduce some risk and we are affectively creating a cache per user.  If there is a really big spike in users, it's going to be quite a lot of replicated and stored data.